src/protocols/bdx/BDXMessageUtils.cpp

```diff
@@ -0,0 +1,408 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      Implements utility methods for working with some complex BDX messages.
+ */
+
+#include <protocols/bdx/BDXMessageUtils.h>
+
+#include <support/BufBound.h>
+#include <support/BufferReader.h>
+#include <support/CodeUtils.h>
+
+#define VERSION_MASK 0x0F
+#define SENDER_DRIVE_MASK 0x10
+#define RECEIVER_DRIVE_MASK 0x20
```

Comment:
should these be Flags instead of Masks?

Reply:
In terms of naming convention? Yes, I think you're right